### PR TITLE
Add appName config option

### PR DESCRIPTION
**Pull request by ironyman**
We have multiple applications pushing logs to one log server and we'd like to differentiate between them in ElasticSearch queries.

### CODE REVIEW

**Comment by id0Sch** [comment]
Hi,
Thanks for taking the time to contribute to this project!
since now we will have both `source` & `appName` - it got me thinking maybe we need something more dynamic, maybe a new config object that will get merged into the output, WDYT?

File: lib/jsonLayout.js [hunk 1]
```diff
@@ -79,6 +79,13 @@ function jsonLayout(config) {
 
     _.assign(output, overlays);
 
+    // Add appName if specified
+    if (_.has(config, 'appName')) {
```
Comment (id0Sch) [inline lib/jsonLayout.js]: do you mind moving this to line 64, just after the `config.source` handling? it will be clearer what gets added.
thanks

**Comment by ironyman** [comment]
Sure, so it would be used like this
```
log4js.configure({
  appenders: {
    out: {
      type: 'console', layout: {
        type: 'json',
        static: {
          appName: 'testapp',
          source: 'development'
        }
      }
    }
  },
  categories: { default: { appenders: ['out'], level: 'info' } }
});
```

**Comment by id0Sch** [comment]
Yeah exactly, so if it's provided we simple merge it onto the object.

**Comment by ironyman** [comment]
Great, I've already pushed the change (in case you didn't notice).

**Comment by id0Sch** [comment]
looks great,
do you mind updating the README + add test case?

**Comment by ironyman** [comment]
I left the mentions of `source` option in the README. What do you think?

**Comment by id0Sch** [comment]
it will be used only for backwards compatibility, i think it's fine.

**Comment by id0Sch** [comment]
@ironyman published, it's available in version 2.1.0
thanks for you time! much appreciated 👏

**Comment by ironyman** [comment]
Thank you.

**Pull request merged**